docs/testing.md

```diff
@@ -67,3 +106,224 @@ nextflow run . \
+```
+
+### **Understanding Your Receipt**
+
+For comprehensive information about interpreting your batch receipt, including:
+- Detailed field descriptions
+- Error message meanings
+- Troubleshooting guidance
+- Best practices for receipt analysis
+
+**👉 See [Receipt Documentation](receipt.md) for complete details**
+
+### **Examining the Specified Output Directory**
+
+Beyond the batch receipt files, your specified output directory (`--outdir`) contains additional important information about the workflow execution:
+
+**Complete Output Structure**:
+Your output directory will contain various subdirectories with intermediate files, logs, and results from each workflow process. This includes:
+- Process-specific input/output files
+- Intermediate data files
+- Process execution logs
+- Pipeline execution reports
+
+**👉 See [Output Documentation](output.md) [**TODO**] for comprehensive details about all output files and directories**
+
+## 📋 **Common Failure Issues**
+
+When running the workflow, you may encounter these common issues across any test scenario:
+
+### **Authentication and Network Issues**
+- **Invalid authentication token**: Expired or incorrect authentication credentials
+- **Network connectivity issues**: Problems connecting to PCGL submission services  
+- **Service availability**: PCGL services temporarily unavailable
+
+### **Biospecimen Entity Issues**
+- **Biospecimen validation errors**: Issues with biospecimen metadata format or content
+- **Entity dependency conflicts**: The provided biospecimen metadata conflict with existing entities
+- **Missing metadata for required entities**: Some biospecimen entities expected but metadata not provided
+- **Biospecimen entity not found**: Expected pre-submitted biospecimen entities don't exist in PCGL
+
+### **Analysis and File Issues**  
+- **Missing workflow metadata**: Required workflow metadata not provided for certain analysis types
+- **Analysis type validation errors**: Analysis metadata doesn't meet requirements for the specified type
+- **File metadata validation failures**: Issues with file metadata format or content
+- **File transfer errors**: Issues uploading genomic files to object storage
+
+### **System Issues**
+- **Resource constraints**: Insufficient system resources during processing
+- **Timeout errors**: Operations exceeding configured time limits
+
+**👉 For detailed troubleshooting steps and solutions, see [Troubleshooting Guide](troubleshooting.md)** [**TODO**]
+
+
+
+## 💡 **Testing Best Practices**
+
+1. **Start with Scenario 1 (No Pre-submitted Biospecimen)**: Verify full workflow functionality including biospecimen entity creation
+2. **Progress to Scenario 2 (All Pre-submitted)**: Test file and analysis only submission pathway
+3. **Test Scenario 3 (Partial Pre-submitted)**: Validate mixed entity state handling
+4. **Review Batch Receipts**: Always check the receipt files after each test run to understand biospecimen entity processing
```

Comment:
- Note from nextflow's end, process will always show as complete because of error redirection and handling. As such, logs and receipts are the best method of debugging.

Reply:
- Processes will automatically fail if the prior process did not successfully complete. One can navigate starting from the latest process backwards to find the origin failing source